Scripts/Core/VRView.cs

```diff
@@ -189,6 +184,70 @@ public void OnEnable()
+            {
+                var cameraGameObject = m_Camera.gameObject;
+                var potentialImageEffects = s_ExistingSceneMainCamera.GetComponents<MonoBehaviour>();
+                var enabledPotentialImageEffects = potentialImageEffects.Where(x => x != null && x.enabled);
+                var targetMethodNames = new [] {"OnRenderImage", "OnPreRender", "OnPostRender"};
+                var bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static;
+                foreach (var potentialImageEffect in enabledPotentialImageEffects)
+                {
+                    var componentInstanceType = potentialImageEffect.GetType();
+                    var targetMethodFound = false;
+                    for (int i = 0; i < targetMethodNames.Length; ++i)
+                    {
+                        // Each of the three checks is performed to catch the various image effect variants I've tested against
+                        // Each check catches a different case that was encountered during testing
+                        // Check isntanced type for target methods
+                        targetMethodFound = componentInstanceType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+
+                        // Check base type for target methods
+                        if (!targetMethodFound)
+                        {
+                            var componentBaseType = componentInstanceType.BaseType;
+                            if (componentBaseType != null)
+                                targetMethodFound = componentBaseType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+                        }
+
+                        // Check nested types for target methods
+                        if (!targetMethodFound)
+                        {
+                            var nestedTypes = componentInstanceType.GetNestedTypes();
+                            foreach (var nestedType in nestedTypes)
+                            {
+                                if (nestedType == null)
+                                    continue;
+
+                                targetMethodFound = nestedType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+                                if (targetMethodFound)
+                                    break;
+                            }
+                        }
+
+                        if (targetMethodFound)
+                            break;
+                    }
+
+                    if (targetMethodFound)
+                    {
+                        try
+                        {
+                            // During testing, some image effects caused Unity to crash when copied
+                            ObjectUtils.CopyComponent(potentialImageEffect, cameraGameObject);
+                        }
+                        catch (Exception e)
+                        {
+                            Console.WriteLine(e);
```

Comment:
Why not Debug.Log? Also, what exceptions? What Unity version? I'm curious about why catching them here would prevent a crash.

Reply:
I'll remove this. After additional testing, a small number of image effects still crash the editor regardless of the try/catch.